pkg/util/kubernetes/lookup.go

```diff
@@ -52,6 +52,17 @@ func LookupConfigmap(ctx context.Context, c client.Client, ns string, name strin
 	return &cm
 }
 
+// LookupResourceVersion will look for any k8s resource with a given name in a given namespace, returning its resource version only.
+// It makes this safe against any resource that the operator is not allowed to inspect.
+func LookupResourceVersion(ctx context.Context, c client.Client, object ctrl.Object) string {
+	if err := c.Get(ctx, ctrl.ObjectKeyFromObject(object), object); err != nil && k8serrors.IsNotFound(err) {
+		return ""
+	} else if err != nil {
+		return ""
```

Comment:
is swallowing the error expected ? if so `k8serrors.IsNotFound(err)` can probably be removed